gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -134,18 +135,9 @@ public Fork(TaskContext taskContext, Object schema, int branches, int index) thr
       buildWriterIfNotPresent();
     }
 
-    this.recordQueue = BoundedBlockingRecordQueue.newBuilder()
-        .hasCapacity(this.taskState.getPropAsInt(
+    this.recordQueue = new LinkedBlockingQueue<Object>(this.taskState.getPropAsInt(
```

Comment:
Yes, I also liked the stats, have to inspect the code more to see if I can get a blocking bahavior out of it. Maybe having Interger.MAX_VALUE as a timeout.